lib/dl_core/dl_core/us_manager/schema_migration/base.py

```diff
@@ -0,0 +1,81 @@
+from copy import deepcopy
+from datetime import datetime
+import logging
+from typing import (
+    Callable,
+    ClassVar,
+)
+
+import attr
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+@attr.s
+class Migration:
+    version: str = attr.ib()
+    name: str = attr.ib()
+    function: Callable[[dict], dict] = attr.ib()
+    id: int = attr.ib(init=False)
+
+    def __attrs_post_init__(self) -> None:
+        self.id = int(datetime.fromisoformat(self.version).timestamp())
+
+    def migrate(self, entry: dict) -> dict:
+        entry = self.function(entry)
+        entry["data"]["schema_version"] = self.version
+        return entry
+
+
+@attr.s
+class BaseEntrySchemaMigration:
+    _MIGRATIONS: ClassVar[list[Migration]] = attr.ib(init=False, factory=list)
+    strict_migration: bool = attr.ib(default=False)
+
+    @classmethod
+    def _collect_migrations(cls) -> list[Migration]:
+        migrations: list[Migration] = []
+        for _cls in cls.__mro__:
```

Comment:
ooof, mb better use `super().` mechanism to iterate over mro?

Reply:
imho, using `__mro__` attribute is much clearer vs `super()` in this case